### PR TITLE
ELY-2049 Add WildFly SecurityManager findAccessDenial trace capability

### PR DESCRIPTION
**Pull request by boris-unckel**
Fixes [ELY-2049](https://issues.redhat.com/browse/ELY-2049)

### CODE REVIEW

File: manager/base/src/main/java/org/wildfly/security/manager/_private/SecurityMessages.java [hunk 1]
```diff
@@ -45,6 +50,10 @@
     @Message(value = "Permission check failed (permission \"%s\" in code source \"%s\" of \"%s\")")
     void accessCheckFailed(Permission permission, CodeSource codeSource, ClassLoader classLoader);
 
+    @LogMessage(level = TRACE)
```
Comment (darranl) [inline manager/base/src/main/java/org/wildfly/security/manager/_private/SecurityMessages.java:53]: It is correct you have updated SecurityMessages to extend BasicLogger but we don't add log message for TRACE level logging, now you have extended BasicLogger you can just call tracef directly.

**Review by darranl — CHANGES_REQUESTED** [review]
Thank you for your contribution, just a small change required as we don't need TRACE messages themselves adding to the SecurityMessages class.

File: manager/base/src/main/java/org/wildfly/security/manager/_private/SecurityMessages.java (hunk 1 above)
Comment (boris-unckel) [reply manager/base/src/main/java/org/wildfly/security/manager/_private/SecurityMessages.java]: I have updated the pull request: The trace has been removed from SecurityMessages and been placed into WFSM directly.

**Comment by boris-unckel** [comment]
@darranl Could you please review again? I'm still in the hope to get the trace capability into regular production of next WF. I have a second change at the same place, but I don't want them to mix up.

**Review by darranl — APPROVED** [review]

**Review by fjuma — APPROVED** [review]

**Comment by boris-unckel** [comment]
@darranl Thank you, but it will be a major and I have to prepare a very good test case first. It will deal with null code sources in protection domains. I'd like to bring [PR 1476](https://github.com/wildfly-security/wildfly-elytron/pull/1467) to your attention, the bouncy castle version.

**Comment by darranl** [comment]
We will review the BouncyCastle PR but really that will only affect the Elytron testsuite, the main thing is the WildFly Core upgrade has gone through for that one so will be included in WildFly 22.

**Pull request merged**